diff --git a/sqlite-wasm/jswasm/sqlite3-bundler-friendly.mjs b/sqlite-wasm/jswasm/sqlite3-bundler-friendly.mjs
index 1e35522cb97757472bd623dd0a332c8640b2d806..1447943319bc9e6732bbfe92098b45f0fbce302b 100644
--- a/sqlite-wasm/jswasm/sqlite3-bundler-friendly.mjs
+++ b/sqlite-wasm/jswasm/sqlite3-bundler-friendly.mjs
@@ -13524,7 +13524,9 @@ var sqlite3InitModule = (() => {
                   return poolUtil;
                 })
                 .catch(async (e) => {
-                  await thePool.removeVfs().catch(() => {});
+                  try {
+                    thePool.pauseVfs();
+                  } catch {}
                   throw e;
                 });
             })
diff --git a/sqlite-wasm/jswasm/sqlite3.js b/sqlite-wasm/jswasm/sqlite3.js
index 45118cbd11d374185291c51a5c76266e9ea1babc..75abe48084c767d238475692bd84537cdf3850a6 100644
--- a/sqlite-wasm/jswasm/sqlite3.js
+++ b/sqlite-wasm/jswasm/sqlite3.js
@@ -13551,7 +13551,9 @@ var sqlite3InitModule = (() => {
                   return poolUtil;
                 })
                 .catch(async (e) => {
-                  await thePool.removeVfs().catch(() => {});
+                  try {
+                    thePool.pauseVfs();
+                  } catch {}
                   throw e;
                 });
             })
diff --git a/sqlite-wasm/jswasm/sqlite3.mjs b/sqlite-wasm/jswasm/sqlite3.mjs
index efd020a12f20938520958a8f359e8114d66c364c..6adb7c2301fa8ad69e4966f0e2cdc578e1f865c3 100644
--- a/sqlite-wasm/jswasm/sqlite3.mjs
+++ b/sqlite-wasm/jswasm/sqlite3.mjs
@@ -13526,7 +13526,9 @@ var sqlite3InitModule = (() => {
                   return poolUtil;
                 })
                 .catch(async (e) => {
-                  await thePool.removeVfs().catch(() => {});
+                  try {
+                    thePool.pauseVfs();
+                  } catch {}
                   throw e;
                 });
             })
